app/assets/javascripts/govuk_publishing_components/load-analytics.js

```diff
@@ -0,0 +1,261 @@
+      'send-money-to-prisoner.service.gov.uk',
+      'signin.service.gov.uk',
+      'sorn.service.gov.uk',
+      'staff.helpwithcourtfees.service.gov.uk',
+      'student-finance.service.gov.uk',
+      'tax.service.gov.uk',
+      'teacherservices.education.gov.uk',
+      'teaching-vacancies.service.gov.uk',
+      'to-visit-or-stay-in-the-uk.homeoffice.gov.uk',
+      'trade-tariff.service.gov.uk',
+      'tribunal-response.employmenttribunals.service.gov.uk',
+      'ukri.org',
+      'update-student-loan-employment-details.service.gov.uk',
+      'vehicle-operator-licensing.service.gov.uk',
+      'vehicleenquiry.service.gov.uk',
+      'viewdrivingrecord.service.gov.uk',
+      'view-and-prove-your-rights.homeoffice.gov.uk',
+      'view-immigration-status.service.gov.uk',
+      'visa-address-update.service.gov.uk',
+      'visas-immigration.service.gov.uk',
+      'your-defra-account.defra.gov.uk'
+    ]
+
+    // Universal Analytics variables
+    window.GOVUK.analyticsVars = window.GOVUK.analyticsVars || {}
+    window.GOVUK.analyticsVars.gaProperty = 'UA-UNSET'
+    window.GOVUK.analyticsVars.gaPropertyCrossDomain = 'UA-145652997-1'
+    window.GOVUK.analyticsVars.primaryLinkedDomains = ['account.gov.uk']
+    window.GOVUK.analyticsVars.linkedDomains = linkedDomains
+
+    if (this.arrayContains(currentDomain, this.productionDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-1'
+    } else if (this.arrayContains(currentDomain, this.stagingDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-20'
+    } else if (this.arrayContains(currentDomain, this.integrationDomains)) {
+      window.GOVUK.analyticsVars.gaProperty = 'UA-26179049-22'
+    } else {
+      window.GOVUK.analyticsVars.gaPropertyCrossDomain = 'UA-UNSET'
+    }
+
+    // Load universal analytics
+    if (typeof window.GOVUK.analyticsInit !== 'undefined') {
+      window.GOVUK.analyticsInit()
+    }
+  },
+
+  loadGa4: function (currentDomain) {
+    currentDomain = currentDomain || window.location.hostname
+    var environment = ''
+
+    // Categorise current environment and set a couple UA variables
+    if (this.arrayContains(currentDomain, this.productionDomains)) {
+      environment = 'production'
+    } else if (this.arrayContains(currentDomain, this.stagingDomains)) {
+      environment = 'staging'
+    } else if (this.arrayContains(currentDomain, this.integrationDomains)) {
+      environment = 'integration'
+    } else if (this.arrayContains(currentDomain, this.developmentDomains) || currentDomain.indexOf('.dev.gov.uk') !== -1) {
+      environment = 'development'
+    }
+
+    // If we recognise the environment (i.e. the string isn't empty), load in GA4
+    if (environment) {
+      // If analytics-ga4.js exists and our detected environment has 'initialiseGA4' set to true, load GA4.
+      if (typeof window.GOVUK.analyticsGa4.init !== 'undefined' && this.ga4EnvironmentVariables[environment].initialiseGA4) {
```

Comment:
I notice that all of the environments have `initialiseGA4` set to `true`, which seems like unnecessary code. I assume this `if` statement is basically checking before proceeding that the environment is valid? Could it not just do a check for `this.ga4EnvironmentVariables[environment]`?

Reply:
Looking at the code more, there's the initial check on line 201 that `environment` exists, are both of these checks necessary? Could they be combined somehow?

Reply:
@andysellick I had added `initialiseGA4` so that we had a method in place to turn GA4 on and off on each environment which didn't require us to have to remove/append code. I'm guessing we might not need that anymore as that was for when we initially deployed GA4. But I was just thinking if we ever needed to turn GA4 off on one environment, having that boolean would be handy. What do you think?

Reply:
That seems like good reason enough. Could you add a comment to that section to highlight this feature?